crates/bevy_core_pipeline/src/auto_exposure/buffers.rs

```diff
@@ -27,13 +28,13 @@ pub(super) struct ExtractedStateBuffers {
 
 pub(super) fn extract_buffers(
     mut commands: Commands,
-    changed: Extract<Query<(Entity, &AutoExposureSettings), Changed<AutoExposureSettings>>>,
+    changed: Extract<Query<(&RenderEntity, &AutoExposureSettings), Changed<AutoExposureSettings>>>,
     mut removed: Extract<RemovedComponents<AutoExposureSettings>>,
 ) {
     commands.insert_resource(ExtractedStateBuffers {
         changed: changed
             .iter()
-            .map(|(entity, settings)| (entity, settings.clone()))
+            .map(|(entity, settings)| (entity.id(), settings.clone()))
```

Comment:
There's quite a lot of noise in this PR due to the need to insert these `RenderEntity::id` calls. This might be a controversial suggestion, but we _can_ implement `WorldQuery` for `RenderEntity` so it still behaves like a `Component`, but also returns the inner `Entity` when queried directly.

<details>
  <summary>unsafe impl WorldQuery for RenderEntity { ... }</summary>
  
  ```rust
/// SAFETY:
/// This implementation delegates to the existing implementation for &RenderEntity
unsafe impl WorldQuery for RenderEntity {
    type Item<'w> = Entity;
    type Fetch<'w> = ReadFetch<'w, RenderEntity>;
    type State = ComponentId;

    fn shrink<'wlong: 'wshort, 'wshort>(item: Entity) -> Entity {
        item
    }

    fn shrink_fetch<'wlong: 'wshort, 'wshort>(fetch: Self::Fetch<'wlong>) -> Self::Fetch<'wshort> {
        <&Self as WorldQuery>::shrink_fetch(fetch)
    }

    #[inline]
    unsafe fn init_fetch<'w>(
        world: UnsafeWorldCell<'w>,
        component_id: &ComponentId,
        last_run: Tick,
        this_run: Tick,
    ) -> ReadFetch<'w, RenderEntity> {
        // SAFETY: This implementation delegates to the existing implementation for &RenderEntity
        unsafe {
            <&Self as WorldQuery>::init_fetch(world, component_id, last_run, this_run)
        }
    }

    const IS_DENSE: bool = <&Self as WorldQuery>::IS_DENSE;

    #[inline]
    unsafe fn set_archetype<'w>(
        fetch: &mut ReadFetch<'w, RenderEntity>,
        component_id: &ComponentId,
        archetype: &'w Archetype,
        table: &'w Table,
    ) {
        // SAFETY: This implementation delegates to the existing implementation for &RenderEntity
        unsafe {
            <&Self as WorldQuery>::set_archetype(fetch, component_id, archetype, table)
        }
    }

    #[inline]
    unsafe fn set_table<'w>(
        fetch: &mut ReadFetch<'w, RenderEntity>,
        component_id: &ComponentId,
        table: &'w Table,
    ) {
        // SAFETY: This implementation delegates to the existing implementation for &RenderEntity
        unsafe {
            <&Self as WorldQuery>::set_table(fetch, component_id, table)
        }
    }

    #[inline(always)]
    unsafe fn fetch<'w>(
        fetch: &mut Self::Fetch<'w>,
        entity: Entity,
        table_row: TableRow,
    ) -> Self::Item<'w> {
        // SAFETY: This implementation delegates to the existing implementation for &RenderEntity
        unsafe {
            <&Self as WorldQuery>::fetch(fetch, entity, table_row).id()
        }
    }

    fn update_component_access(
        component_id: &ComponentId,
        access: &mut FilteredAccess<ComponentId>,
    ) {
        <&Self as WorldQuery>::update_component_access(component_id, access)
    }

    fn init_state(world: &mut World) -> ComponentId {
        <&Self as WorldQuery>::init_state(world)
    }

    fn get_state(components: &Components) -> Option<Self::State> {
        <&Self as WorldQuery>::get_state(components)
    }

    fn matches_component_set(
        state: &ComponentId,
        set_contains_id: &impl Fn(ComponentId) -> bool,
    ) -> bool {
        <&Self as WorldQuery>::matches_component_set(state, set_contains_id)
    }
}

/// SAFETY: `Self` is the same as `Self::ReadOnly`
unsafe impl QueryData for RenderEntity {
    type ReadOnly = Self;
}

/// SAFETY: access is read only
unsafe impl ReadOnlyQueryData for RenderEntity {}
  ```
  
</details>

This would then allow for render systems to query for `RenderEntity` and directly receive the inner `Entity`, reducing the noise in this PR. Crucially, this doesn't conflict with the existing `derive(Component)`, so you can still query for `&RenderEntity`, the mutable version, use it in filters, etc.